pkg/auth/providers/publicapi/login.go

```diff
@@ -87,6 +87,9 @@ func (h *loginHandler) login(actionName string, action *types.Action, request *t
 			HttpOnly: true,
 		}
 		http.SetCookie(w, tokenCookie)
+		w.WriteHeader(http.StatusNoContent)
```

Comment:
How does this fit with the rest of this pr? Not for or against the change, just not quite seeing how it fits in

Reply:
So because I used Teapot status as a canary - this was something I noticed in the audit log as being a 418 teapot status. Meaning it's something that got past our 200 default if `Write` is called before `WriteHeader` (which matches other core go http libraries logic).

Given that I felt it was probably best to correct this API response path rather than adjust the logic further. Since even if we did catch it with the above mentioned logic `200 OK` works but is not strictly the most correct 20x status code.

Because this API response path only seems to care about setting cookie w/o need for response body - technically returning a `204 No Content` is more accurate. And restored login page functionality.

Reply:
Might be best to get Collie's approval on this since this can affect UI and maybe CLI, terraform. It also makes me a bit nervous if say a user is using `/v3` in their tool and check to see if they receive exactly 200, that would break them.

Reply:
Yeah I think you caught up now.

Just for the record, if anyone else is curious, the gist is that even when we mimic stdlib behaviour of `Write`, sometimes a `418` code was seen. I took that as a sign that a response was sent that never called `Write`.

I'll go double check that I didn't overlook something in stdlib about Flush that maybe covers a similar area. Since I am curious now what code sets the status on those when audit log is not enabled 🤔